components/suggest/src/db.rs

```diff
@@ -128,63 +128,72 @@ impl<'a> SuggestDao<'a> {
 
     /// Fetches suggestions that match the given query from the database.
     pub fn fetch_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
-        if let Some(suggestion) = self.fetch_yelp_suggestion(query)? {
-            return Ok(vec![suggestion]);
-        }
-
-        let keyword_lowercased = &query.keyword.to_lowercase();
-        let (keyword_prefix, keyword_suffix) = split_keyword(keyword_lowercased);
         let suggestions_limit = query.limit.unwrap_or(-1);
+        let suggestions =
```

Comment:
I think we can dedup providers before running them though just in case.

```rs
let unique_providers = query.providers.iter().collect::<HashSet<_>>();
unique_providers.iter().try_fold(...)
```

Reply:
Let's add a test case for this :)